docs/design/disqualification.md

```diff
@@ -0,0 +1,77 @@
+
+Nodes can be disqualified due to audit failures, failures to return data during repair, or poor uptime statistics.
+
+## Main Objectives
+
+1. To not send new data to nodes which are not behaving as desired.
+2. Efficiently identify and ignore these 'bad' nodes during various node selection and statistics processes.
+
+## Background
+
+The whitepaper section 4.13 talks about containment mode as follows:
+
+> The filtering system is the third subsystem; it blocks bad storage nodes from participating. In addition to simply not having done a suficient proof of work, certain actions a storage node can take are disqualifying events. The reputation system will be used to filter these nodes out from future uploads, regardless of where the node is in the vetting process. Actions that are disqualifying include: failing too many audits; failing to return data, with reasonable speed; and failing too many uptime checks.
+> If a storage node is disqualified, that node will no longer be selected for future data storage and the data that node stores will be moved to new storage nodes. Likewise, if a client attempts to download a piece from a storage node that the node should have stored and the node fails to return it, the node will be disqualified. Importantly, storagenodes will be allowed to reject and fail _put_ operations without penalty, as nodes will be allowed to choose which Satellite operators to work with and which data to store.
+
+'Failing to return data' is clarified to mean during an audit or a repair. Failure to return data to uplinks is specifically excluded, as this would imply a robust system of trust that does not currently exist.
+
+'Regardless of... the vetting process' is highlighted to show that both vetted and new nodes may be disqualified.
+
+> After a storage node is disqualified, the node must go back through the entire vetting process again. If the node decides to start over with a brand-new identity, the node must restart the vetting process from the beginning (in addition to generating a new nodeID via the proof of work system). This strongly disincentivizes storage nodes from being cavalier with their reputation.
+
+Further, the node will be demonetized.
+
+> Provided the storage node hasn’t been disqualified, the storage node will be paid by the Satellite for the data it has stored over the course ofthe month, per the Satellite’s records.
+
+A disqualified SNO should quickly stop particiapting with a satellite it is disqualified and demonetized on.  It may remain in Kademlia and potentially found during Overlay Cache Discover, as the kademlia network supports multiple satellites.
+
+One option that may be allowed for disqualified storage nodes is a Graceful Exit.  [Storage Node Payment and Incentives for V3](https://docs.google.com/document/d/1-Pxzk-ad-0QtF6nnTwfgzk8e_-XbBNSDxuRvnbd0QL8/edit#heading=h.rz1ehm5mbeuz) describes this feature:
```

Comment:
I think this may trigger some long discussion whether or not we should allow Graceful Exit for nodes that are already disqualified. Perhaps, we should leave it for the future design doc for Graceful Exit.

Reply:
I'll add a 'to be determined' statement of some sort here.

Reply:
updated